node/api.go

```diff
@@ -58,14 +59,30 @@ func (api *PrivateAdminAPI) AddPeer(url string) (bool, error) {
 }
 
 // StartRPC starts the HTTP RPC API server.
-func (api *PrivateAdminAPI) StartRPC(host string, port int, cors string, apis string) (bool, error) {
+func (api *PrivateAdminAPI) StartRPC(host *string, port *rpc.HexNumber, cors *string, apis *string) (bool, error) {
 	api.node.lock.Lock()
 	defer api.node.lock.Unlock()
 
 	if api.node.httpHandler != nil {
 		return false, fmt.Errorf("HTTP RPC already running on %s", api.node.httpEndpoint)
 	}
-	if err := api.node.startHTTP(fmt.Sprintf("%s:%d", host, port), api.node.rpcAPIs, strings.Split(apis, ","), cors); err != nil {
+
+	if host == nil {
+		host = new(string)
+		*host = common.DefaultHTTPHost
```

Comment:
All in all I think this `common.DefaultHTTPHost` and its counterparts below (`common.DefaultHTTPPort` and `rpc.DefaultHTTPApis`) are a bit unexpected. I think the only sane default value we should set is whatever was configured via the CLI params. If the user set `--rpcapi=admin`, I'd expect that to be te auto enabled one and not some hard coded value (makes defaults much much less useful).